servers/src/mining/mine_block.rs

```diff
@@ -265,15 +266,48 @@ fn get_coinbase(
 /// Call the wallet API to create a coinbase output for the given block_fees.
 /// Will retry based on default "retry forever with backoff" behavior.
 fn create_coinbase(dest: &str, block_fees: &BlockFees) -> Result<CbData, Error> {
-	let url = format!("{}/v1/wallet/foreign/build_coinbase", dest);
-	match api::client::post(&url, None, &block_fees) {
-		Err(e) => {
-			error!(
-				"Failed to get coinbase from {}. Is the wallet listening?",
-				url
-			);
-			Err(Error::WalletComm(format!("{}", e)))
+	let url = format!("{}/v2/foreign", dest);
+	let req_body = json!({
+		"jsonrpc": "2.0",
+		"method": "build_coinbase",
+		"id": 1,
+		"params": {
+			"block_fees": block_fees
 		}
-		Ok(res) => Ok(res),
+	});
+
+	trace!("Sending build_coinbase request: {}", req_body);
+	let req = api::client::create_post_request(url.as_str(), None, &req_body)?;
+	let res: String = api::client::send_request(req).map_err(|e| {
+		let report = format!(
+			"Failed to get coinbase from {}. Is the wallet listening? {}",
+			dest, e
+		);
+		error!("{}", report);
+		Error::WalletComm(report)
+	})?;
+
+	let res: Value = serde_json::from_str(&res).unwrap();
```

Comment:
Wouldn't be cleaner to create a type for the response (I assume it's Result-like) instead of dealing with raw json Value?

Reply:
It just seemed simpler for this one-off in the code to do this way, also had the notion of making it obvious what the json structure is underneath for anyone looking at the code. If it were a full wallet API client, I'd say definitely, yes.